legacy/reconcile_imageinfo.go

```diff
@@ -0,0 +1,118 @@
+package legacy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/entgigi/upgrade-operator.git/common"
+	"github.com/entgigi/upgrade-operator.git/service"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type ImageInfoEntry struct {
+	Version        string `json:"version"`
+	ExecutableType string `json:"executable-type"`
+	Registry       string `json:"registry"`
+	Organization   string `json:"organization"`
+	Repository     string `json:"repository"`
+}
+
+func (r *LegacyReconcileManager) ReconcileImageInfo(ctx context.Context, req ctrl.Request, appImages service.EntandoAppImages) error {
+	r.Log.Info("Starting ImageInfo configMap reconciliation flow")
+
+	configMap := &corev1.ConfigMap{}
+	var imageInfo types.NamespacedName = types.NamespacedName{
+		Name:      common.ImageInfoConfigMap,
+		Namespace: req.NamespacedName.Namespace,
+	}
+	if err := r.Client.Get(ctx, imageInfo, configMap); err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("ImageInfo configMap not found")
+			return nil
+		}
+		r.Log.Error(err, "Error get ImageInfo configMap")
+		return err
+	}
+
+	for key, value := range configMap.Data {
+		switch {
+		case key == "app-builder-6-4":
```

Comment:
why not using `switch(key)` instead of repeating `key ==`?

Reply:
my bad i didn't know very well switch in golang

Reply:
done